app/restore_to.go

```diff
@@ -0,0 +1,145 @@
+package app
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/longhorn-engine/replica"
+	"github.com/rancher/longhorn-engine/util"
+	"github.com/urfave/cli"
+)
+
+const DefaultImageFormat = "qcow2"
+
+var SupportedImageFormats = []string{
+	"rbd",
+	"host_cdrom",
+	"blkdebug",
+	"qcow",
+	"host_device",
+	"vpc",
+	"qcow2",
+	"cloop",
+	"vdi",
+	"sheepdog",
+	"qed",
+	"nbd",
+	"tftp",
+	"vvfat",
+	"ftp",
+	"ftps",
+	"https",
+	"dmg",
+	"http",
+	"vmdk",
+	"iscsi",
+	"parallels",
+	"raw",
+	"bochs",
+	"quorum",
+	"null-aio",
+	"null-co",
+	"vhdx",
+	"blkverify",
+	"file",
+}
+
+func RestoreToCmd() cli.Command {
+	return cli.Command{
+		Name: "restore-to",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "backing-file",
+				Usage: "filepath or dirpath containing exactly one qcow2 backing file",
+			},
+			cli.StringFlag{
+				Name:  "backup-url",
+				Usage: "backup URL to be published",
+			},
+			cli.StringFlag{
+				Name:  "image-format",
+				Usage: "format of image to produce",
+				Value: DefaultImageFormat,
+			},
+			cli.StringFlag{
+				Name:  "output-file",
+				Usage: "filepath to write the resulting image to",
+				Value: "/image/base.qcow2",
```

Comment:
Output to the current directory by default. `volume.qcow2`.

Check and make sure `raw` works as well.